crates/revm/src/db/states/bundle_state.rs

```diff
@@ -7,6 +7,148 @@ use revm_interpreter::primitives::{
+
+    reverts: HashSet<(usize, B160)>,
+    revert_account: HashMap<(usize, B160), Option<AccountInfo>>,
+    revert_storage: HashMap<(usize, B160), Vec<(U256, U256)>>,
+
+    contracts: HashMap<B256, Bytecode>,
+}
+
+impl BundleBuilder {
+    pub fn new() -> Self {
+        BundleBuilder::default()
+    }
+
+    pub fn state_address(mut self, address: B160) -> Self {
+        self.states.insert(address);
+        self
+    }
+
+    pub fn state_original_account_info(mut self, address: B160, original: AccountInfo) -> Self {
+        self.states.insert(address);
+        self.state_original.insert(address, original);
+        self
+    }
+
+    pub fn state_present_account_info(mut self, address: B160, present: AccountInfo) -> Self {
+        self.states.insert(address);
+        self.state_present.insert(address, present);
+        self
+    }
+
+    pub fn state_storage(mut self, address: B160, storage: HashMap<U256, (U256, U256)>) -> Self {
+        self.states.insert(address);
+        self.state_storage.insert(address, storage);
+        self
+    }
+
+    pub fn revert_address(mut self, block_index: usize, address: B160) -> Self {
+        self.reverts.insert((block_index, address));
+        self
+    }
+
+    pub fn revert_account_info(
+        mut self,
+        block_index: usize,
+        address: B160,
+        account: Option<AccountInfo>,
+    ) -> Self {
+        self.reverts.insert((block_index, address));
+        self.revert_account.insert((block_index, address), account);
+        self
+    }
+
+    pub fn revert_storage(
+        mut self,
+        block_index: usize,
+        address: B160,
+        storage: Vec<(U256, U256)>,
+    ) -> Self {
+        self.reverts.insert((block_index, address));
+        self.revert_storage.insert((block_index, address), storage);
+        self
+    }
+
+    pub fn contract(mut self, address: B256, bytecode: Bytecode) -> Self {
+        self.contracts.insert(address, bytecode);
+        self
+    }
+
+    pub fn build(mut self) -> BundleState {
+        let state = self
+            .states
+            .into_iter()
+            .map(|address| {
+                let storage = self
+                    .state_storage
+                    .remove(&address)
+                    .map(|s| {
+                        s.into_iter()
+                            .map(|(k, (o_val, p_val))| (k, StorageSlot::new_changed(o_val, p_val)))
+                            .collect()
+                    })
+                    .unwrap_or_default();
+                let bundle_account = BundleAccount::new(
+                    self.state_original.remove(&address),
+                    self.state_present.remove(&address),
+                    storage,
+                    AccountStatus::Changed,
+                );
+                (address, bundle_account)
+            })
+            .collect();
+
+        let block_reverts = self
+            .reverts
```

Comment:
You need to be careful to fill the gaps and insert empty vec if there is no reverts for that block.

For example:
`[[Block1Revert,[],[Block3Revert]]` if block1 and block3 reverts are inserted, block2 empty revert should be present.

Reply:
Ah yes, thanks for the catch, I missed it.